stress_test/controller.py

```diff
@@ -0,0 +1,349 @@
+        else:
+            logging.info('[start_controller] Controller already started.')
+
+
+    def stop_controller(ssh_client=None):
+        """Wrapper to the controller stop handler
+
+        :param ssh_client : SSH client provided by paramiko to run the command
+        :type ssh_client: paramiko.SSHClient
+        """
+
+        if check_controller_status(ssh_client) == '1':
+            logging.info('[stop_controller] Stopping controller.')
+            common.command_exec_wrapper(
+                [self.stop_handler],
+                '[controller_stop_handler]', ssh_client)
+            util.process.wait_until_process_finishes(self.cpid, ssh_client)
+        else:
+            logging.info('[stop_controller] Controller already stopped.')
+
+
+    def wait_until_controller_listens(interval_ms, ssh_client=None):
+        """ Waits for controller to start listening on specified port.
+
+        :param interval_ms: milliseconds to wait (in milliseconds).
+        :param ssh_client : SSH client provided by paramiko to run the command
+        :raises Exception: If controller fails to start or if another process
+        listens on controllers port.
+        :rtype int
+        :type interval_ms: int
+        :type ssh_client: paramiko.SSHClient
+        """
+
+        timeout = time.time() + (float(interval_ms) / 1000)
+        while time.time() < timeout:
+            time.sleep(1)
+            gpid = util.process.getpid_listeningonport(self.port, ssh_client)
+            logging.info('Returned pid listening on port {0}: {1}'.
+                          format(self.port, gpid))
+            if gpid == 0:
+                raise Exception('Another controller seems to have started in the '
+                                'meantime. Exiting...')
+
+        raise Exception('Controller failed to start within a period of {0} '
+                        'minutes'.format(timeout))
+
+
+    def wait_until_controller_up_and_running(interval_ms, ssh_client=None):
+        """ Waits for controller status to become 1 (started).
+
+        :param interval_ms: milliseconds to wait (in milliseconds).
+        :param controller_status_handler: filepath to the controller status handler
+        :raises Exception: If controller fails to start or if another process
+        listens on controllers port.
+        :type interval_ms: int
+        :type ssh_client: paramiko.SSHClient
+        """
+
+        timeout = time.time() + (float(interval_ms) / 1000)
+        while time.time() < timeout:
+            time.sleep(1)
+            if check_controller_status(ssh_client) == '1':
+                return
+
+        raise Exception('Controller failed to start. '
+                        'Status check returned 0 after trying for {0} seconds.'.
+                        format(float(interval_ms) / 1000))
+
+
+
+class ODL(Controller):
+    def __init__(self,ctrl_base_dir,test_config):
+        Controller.__init__(self,test_config)
```

Comment:
better practice: use `super` to initialize parent classes (see Item 25 from Effective Python book)